### PR TITLE
Confidential surjection proof

### LINKED ISSUE #66: [Confidential] Surjection proof
We need to expose a `SurjectionProof` method in our *confidential* package that expects the asset id of the output to blind, the output asset blinding factor, all the asset commitments and blinding factors of the *unblinded* output used as inputs in the transaction and a random 32-byte array seed and returns a surjection proof.

This method uses the params to initialize and generate a surjection proof with `secp256k1_surjectionproof_inititialize` and `secp256k1_surjectionproof_inititialize`, and finally returns the serialized raw bytes thanks to `secp256k1_surjectionproof_serialize`.

[This](https://github.com/vulpemventures/liquidjs-lib/blob/97673986c4233dee61f98dc59b5256ed2a8f313d/ts_src/confidential.ts#L133) can be used as reference implementation, while [here](https://github.com/vulpemventures/liquidjs-lib/blob/97673986c4233dee61f98dc59b5256ed2a8f313d/ts_src/psbt.ts#L786) you can see how this is used by the blinder role of the liquidjs-lib's *pset* module.

### PR DESCRIPTION
**Pull request by sekulicd**
Before there was no option to calculate Surjection Proof.

After the method SurjectionProof is added, the user can calculate surjection proof used in a confidential transaction.

It closes #66  .

Please @altafan can you review this?

### CODE REVIEW

File: confidential/confidential.go [hunk 1]
```diff
@@ -125,3 +125,65 @@ func ValueCommitment(value uint64, generator []byte, factor []byte) (result [33]
...
+	InputAssetBlindingFactors [][]byte
+	Seed                      []byte
+}
+
+//SurjectionProof method generates surjection proof
+func SurjectionProof(input SurjectionProofInput) ([]byte, error) {
+	ctx, _ := secp256k1.ContextCreate(secp256k1.ContextBoth)
+	defer secp256k1.ContextDestroy(ctx)
+
+	outputGenerator, err := secp256k1.GeneratorGenerateBlinded(ctx, input.OutputAsset, input.OutputAssetBlindingFactor)
```
Comment (altafan) [inline confidential/confidential.go:142]: Take a look at [this comment](https://github.com/vulpemventures/go-elements/pull/72/files#r434452789) about code style guides.

File: confidential/confidential.go [hunk 2]
```diff
@@ -125,3 +125,65 @@ func ValueCommitment(value uint64, generator []byte, factor []byte) (result [33]
...
+	if err != nil {
+		return nil, err
+	}
+
+	inputGenerators := make([]secp256k1.Generator, 0)
+	for i, v := range input.InputAssets {
+		gen, err := secp256k1.GeneratorGenerateBlinded(ctx, v, input.InputAssetBlindingFactors[i])
```
Comment (altafan) [inline confidential/confidential.go:149]: same here

File: confidential/confidential.go [hunk 3]
```diff
@@ -125,3 +125,65 @@ func ValueCommitment(value uint64, generator []byte, factor []byte) (result [33]
...
+		nInputsToUse = len(input.InputAssets)
+	}
+
+	fixedOutputTag, err := secp256k1.FixedAssetTagParse(input.OutputAsset)
+	if err != nil {
+		return nil, err
+	}
+
+	maxIterations := 100
+	proof, inputIndex, err := secp256k1.SurjectionProofInitialize(ctx, fixedInputTags, nInputsToUse, *fixedOutputTag, maxIterations, input.Seed)
```
Comment (altafan) [inline confidential/confidential.go:178]: and here

File: confidential/confidential.go [hunk 4]
```diff
@@ -125,3 +125,65 @@ func ValueCommitment(value uint64, generator []byte, factor []byte) (result [33]
...
+	if err != nil {
+		return nil, err
+	}
+
+	err = secp256k1.SurjectionProofGenerate(ctx, proof, inputGenerators, *outputGenerator, inputIndex, input.InputAssetBlindingFactors[inputIndex], input.OutputAssetBlindingFactor)
```
Comment (altafan) [inline confidential/confidential.go:183]: and here

File: confidential/confidential.go (hunk 1 above)
Comment (sekulicd) [reply confidential/confidential.go:142]: Done

File: confidential/confidential.go (hunk 2 above)
Comment (sekulicd) [reply confidential/confidential.go:149]: Done

File: confidential/confidential.go (hunk 3 above)
Comment (sekulicd) [reply confidential/confidential.go]: Done

File: confidential/confidential.go (hunk 4 above)
Comment (sekulicd) [reply confidential/confidential.go]: Done

**Pull request merged**